src/wtr-config.js

```diff
@@ -0,0 +1,106 @@
+import { defaultReporter } from '@web/test-runner';
+import { playwrightLauncher } from '@web/test-runner-playwright';
+import { visualDiff } from './visual-diff-plugin.js';
+import { visualDiffReporter } from './visual-diff-reporter.js';
+
+const DEFAULT_PATTERN = type => `./**/*.${type}.js`;
+const DEFAULT_VDIFF = true;
+const DEFAULT_BROWSERS = ['chromium', 'firefox', 'webkit'];
+
+export function getBrowsers(browsers = DEFAULT_BROWSERS) {
+
+	console.assert(Array.isArray(browsers), 'browsers must be an array');
+
+	return browsers.map((b) => playwrightLauncher({
+		product: b,
+		createBrowserContext: ({ browser }) => browser.newContext({ deviceScaleFactor: 2, reducedMotion: 'reduce' })
+	}));
+}
+
+function getVisualDiffGroup(pattern) {
+	return {
+		name: 'vdiff',
+		files: pattern('vdiff'),
+		browsers: getBrowsers(['chromium']),
+		testRunnerHtml: testFramework =>
+			`<html lang="en">
+				<head>
+					<link rel="preload" href="https://s.brightspace.com/lib/fonts/0.5.0/assets/Lato-400.woff2" as="font" type="font/woff2" crossorigin>
+					<link rel="preload" href="https://s.brightspace.com/lib/fonts/0.5.0/assets/Lato-700.woff2" as="font" type="font/woff2" crossorigin>
+					<style>
+						html {
+							font-size: 20px;
+						}
+						body {
+							background-color: #ffffff;
+							margin: 0;
+							padding: 30px;
+						}
+						body[data-theme="dark"] {
+							background-color: #000000;
+						}
+						body[data-theme="translucent"] {
+							background: repeating-linear-gradient(45deg, #606dbc, #606dbc 10px, #465298 10px, #465298 20px);
+						}
+					</style>
+					<script type="module" src="./components/typography/typography.js"></script>
+				</head>
+				<body class="d2l-typography">
+					<script type="module" src="${testFramework}"></script>
+				</body>
+			</html>`
+	};
+}
+
+export function createConfig({
+	pattern = DEFAULT_PATTERN,
+	vdiff = DEFAULT_VDIFF,
+	timeout,
+	...passthroughConfig
+} = {}) {
+
+	console.assert(typeof pattern === 'function', 'pattern must be a function');
```

Comment:
I haven't seen this pattern before... I don't think it stops execution like a `throw` would? I'd probably just throw `TypeError`s in these scenarios.